webgpu.h

```diff
@@ -1354,10 +1353,10 @@ typedef void (*WGPUProcShaderModuleSetLabel)(WGPUShaderModule shaderModule, char
 
 // Procs of Surface
 typedef WGPUTextureFormat (*WGPUProcSurfaceGetPreferredFormat)(WGPUSurface surface, WGPUAdapter adapter);
-
-// Procs of SwapChain
-typedef WGPUTextureView (*WGPUProcSwapChainGetCurrentTextureView)(WGPUSwapChain swapChain);
-typedef void (*WGPUProcSwapChainPresent)(WGPUSwapChain swapChain);
+typedef void (*WGPUProcSurfaceConfigure)(WGPUSurface surface, WGPUDevice device, WGPUSurfaceConfiguration const * configuration);
+typedef void (*WGPUProcSurfaceUnconfigure)(WGPUSurface surface);
+typedef WGPUTexture (*WGPUProcGetCurrentTexture)(WGPUSurface surface, bool * suboptimal);
```

Comment:
nit: Maybe it could be a separate call, but also optionally return some optimal parameters like the size of the texture?

Reply:
wouldn't separate calls to `wgpuTextureGetWidth` & `wgpuTextureGetHeight` on the returned `WGPUTexture` suffice?

Reply:
I don't think so. The texture you get should have exactly the size defined in `WGPUSurfaceConfiguration` otherwise the application doesn't really control when the resize happens and needs to be ready to resize other rendertargets any time. (plus if they forget to check then they'll get errors leading to rendering a black screen). At least in Dawn we have manual test that the resizing work, and it is implemented with a blit in the backends if it is needed (of course optimal would return false ^^)

Reply:
ack